### PR TITLE
Input connected + add possibility to set a label from the input props

### PR DESCRIPTION
**Pull request by MichelML**
@sbelzile-Coveo this doesn't break the examples in the docs, but you tell me if this could potentially break something on your side.

demo: https://coveo.github.io/react-vapor/input-connected/ , search for `Inputs list` in the page for quick access

### CODE REVIEW

File: src/animations/SlideY.spec.tsx [hunk 1]
```diff
@@ -3,49 +3,49 @@ import { shallow } from 'enzyme';
 import { SlideY, onEntering, onExiting } from './SlideY';
```
Comment (MichelML) [inline src/animations/SlideY.spec.tsx]: ignore this file, this is out of scope

File: src/components/input/InputConnected.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    value: input && input.value,
+    disabled: input && input.disabled,
+  };
+};
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: IInputOwnProps): IInputDispatchProps => ({
+  onRender: (value: string = '', valid = true, disabled = false) => {
+    const validOnMount = ownProps.validateOnMount
+      && ownProps.validate
+      && ownProps.validate(value);
```
Comment (sroy3) [inline src/components/input/InputConnected.tsx]: Aren't you already getting this result from the `valid` parameter? You do the same thing on line 60 of `Input.tsx` where you call `onRender`.

File: src/components/input/InputConnected.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    dispatch(addInput(ownProps.id, value, validOnMount, disabled));
+  },
+  onDestroy: () => dispatch(removeInput(ownProps.id)),
+  onChange: (value: string) => {
+    const validOnChange = ownProps.validateOnChange
+      && ownProps.validate
+      && ownProps.validate(value);
```
Comment (sroy3) [inline src/components/input/InputConnected.tsx]: You should move this to the `Input.tsx` file like you have for the `validOnMount`. This way it can still be used when not using the connected component.

File: src/components/input/tests/InputConnected.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  const mountComponentWithProps = (props: IInputProps) => mount(
+    <Provider store={store}>
+      <InputConnected {...props} />
+    </Provider>,
+    { attachTo: document.getElementById('App') },
+  );
+
+  it('should mount without errors in various props scenarios', () => {
+    expect(() => {
+      mountComponentWithProps({});
```
Comment (sroy3) [inline src/components/input/tests/InputConnected.spec.tsx]: You could set `{}` as the default and send nothing here.

**Comment by MichelML** [comment]
@sbelzile-Coveo this is somewhat good news...let me know if ever you discover something more major

File: src/components/datePicker/tests/DatesSelection.spec.tsx [hunk 5]
```diff
@@ -222,7 +222,7 @@ describe('Date picker', () => {
       }).not.toThrow();
 
       datesSelection.setProps({ onBlur: onBlurSpy });
-      (datesSelection.find('input').get(0) as React.HTMLAttributes<HTMLButtonElement>).value = DateUtils.getDateWithTimeString(new Date());
+      (datesSelection.find('input').get(0) as any).value = DateUtils.getDateWithTimeString(new Date());
```
[the remark this answers is not in the document]
Comment (MichelML) [reply src/components/datePicker/tests/DatesSelection.spec.tsx]:
```
      TS2339: Property 'value' does not exist on type 'HTMLAttributes<HTMLInputElement>'.
```

tried a couple of unsuccessful other types before using any

[the remark this answers is not in the document]
Comment (sroy3) [reply src/components/datePicker/tests/DatesSelection.spec.tsx]: Shouldn't it be `HTMLInputElement`?

[the remark this answers is not in the document]
Comment (MichelML) [reply src/components/datePicker/tests/DatesSelection.spec.tsx]: tried it, same error

**Comment by MichelML** [comment]
@sroy3 for the validateOnMount, there could be a potential case in ML, we'll have a new feature which is to give a name to a Machine Learning Model, initially all machine learning models won't have a name, so we may prompt them to add one the first time they arrive in their machine learning model configuration in the ML UI.

File: src/components/input/InputConnected.tsx (hunk 2 above)
Comment (MichelML) [reply src/components/input/InputConnected.tsx]: done 👍

File: src/components/input/tests/InputConnected.spec.tsx (hunk 4 above)
Comment (MichelML) [reply src/components/input/tests/InputConnected.spec.tsx]: done 👍

File: src/components/input/InputConnected.tsx (hunk 3 above)
Comment (MichelML) [reply src/components/input/InputConnected.tsx]: done 👍 , initially i wanted to put them in InputConnected only but you surely have a good point here

File: src/components/datePicker/tests/DatesSelection.spec.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (sroy3) [reply src/components/datePicker/tests/DatesSelection.spec.tsx]: I think it is because it is a `HTMLInputElement` and not a `React.HTMLAttributes<HTMLInputElement>`

[the remark this answers is not in the document]
Comment (MichelML) [reply src/components/datePicker/tests/DatesSelection.spec.tsx]: Tried it too, but says you can't convert a ReactElement to an HTMLInputElement, tried with ReactElement<HTMLInputElement> too but it's not working, there's always something.
